File: core/src/main/java/org/selyu/smp/core/item/impl/HellstonePickaxeItem.java
package org.selyu.smp.core.item.impl;

import org.bukkit.Material;
import org.jetbrains.annotations.NotNull;
import org.selyu.smp.core.item.CustomItemType;
import org.selyu.smp.core.item.DurableCustomItem;
import org.selyu.smp.core.item.recipe.Recipe;
import org.selyu.smp.core.item.recipe.ShapedRecipeBuilder;

import static org.selyu.smp.core.util.DurabilityUtil.HELLSTONE_INGOT;

public final class HellstonePickaxeItem extends DurableCustomItem {
    public HellstonePickaxeItem() {
        super(CustomItemType.HELLSTONE_PICKAXE, Material.IRON_PICKAXE, 1, HELLSTONE_INGOT * 3);
    }

    @NotNull
    @Override
    public String getDisplayName() {
        return "&fHellstone Pickaxe";
    }

    @Override
    public @NotNull Recipe getRecipe() {
        return new ShapedRecipeBuilder(this)
                .shape(
                        'i', 'i', 'i',
                        '_', 's', '_',
                        '_', 's', '_'
                )
                .ingredient('i', CustomItemType.HELLSTONE_INGOT)
                .ingredient('s', Material.STICK)
                .build();
    }
}

File: core/src/main/kotlin/org/selyu/smp/core/item/impl/HellstonePickaxeItem.kt
package org.selyu.smp.core.item.impl

import org.bukkit.Material
import org.selyu.smp.core.item.CoreItemType
import org.selyu.smp.core.item.DurableCoreItem
import org.selyu.smp.core.item.recipe.CoreRecipe
import org.selyu.smp.core.item.recipe.key.impl.InternalNameRecipeKey
import org.selyu.smp.core.item.recipe.key.impl.MaterialRecipeKey

class HellstonePickaxeItem : DurableCoreItem("hellstone_pickaxe", CoreItemType.PICKAXE, Material.IRON_PICKAXE, 1, 1000) {
    override fun getDisplayName(): String = "&fHellstone Pickaxe"

    override fun getRecipe(): CoreRecipe? = CoreRecipe(this).also {
        it.addKey(InternalNameRecipeKey("hellstone_ingot", 3))
        it.addKey(MaterialRecipeKey(Material.STICK, 2))
    }
}